### PR TITLE
Allow for configurable target-dir

### PR DESCRIPTION
**Pull request by RXminuS**
We have lots of projects in a monorepo and as such we have a shared `dist/cargo` directory that was incompatible.

Could be improved by simply reading this configuration from the Rust config file as well.

### CODE REVIEW

File: src/common.ts [hunk 1]
```diff
@@ -12,6 +12,8 @@ process.on("uncaughtException", (e) => {
 });
 
 const cwd = core.getInput("working-directory");
+//todo: this could be read from .cargo config file directly
+const targetDir = core.getInput("target-dir") || "./target";
```
Comment (Swatinem) [inline src/common.ts:16]: Can you add this to the readme and the action description file as well?

**Review by Swatinem — COMMENTED** [review]
lgtm.

So this is https://doc.rust-lang.org/cargo/reference/config.html#buildtarget-dir right?

I’m not too keen to implement lookup and parsing in JS, so going with an input of the action is fine I guess.

Can you remove your changes to dist files and package-lock? I will update dependencies and run the build before publishing myself.

**Comment by RXminuS** [comment]
Yes, will try to do so soon. Currently on flights.

**Comment by vedhavyas** [comment]
Any chance this could be merged ?
Our project could definitely use this

**Comment by Swatinem** [comment]
Sorry for the delay. I was waiting for you to revert the changes to unrelated files (and dist files), those I would like to just regenerate myself.
Either way, I can also pull just the relevant changes from you, and merge the rest later today manually.

**Comment by vedhavyas** [comment]
@Swatinem I think @RXminuS need to do that but if you could do that and make a release, that would be amazzzing!!

**Pull request closed without merging**